src/components/ui/button.tsx

```diff
@@ -35,13 +35,47 @@ export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
     VariantProps<typeof buttonVariants> {
   asChild?: boolean;
+  pending?: boolean;
 }
 
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ className, variant, size, asChild = false, ...props }, ref) => {
+  ({ className, variant, size, asChild = false, pending = false, children, ...props }, ref) => {
     const Comp = asChild ? Slot : 'button';
     return (
-      <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />
+      <Comp
+        className={cn(
+          buttonVariants({ variant, size, className }),
+          'relative',
+          pending && 'pointer-events-none opacity-50',
+        )}
+        ref={ref}
+        {...props}>
+        {pending && (
+          <div className="absolute inset-0 flex items-center justify-center">
```

Comment:
<img width="672" alt="image" src="https://github.com/user-attachments/assets/1936ab23-57d1-496e-8d05-1b154625dc7e">

버튼 크기에 따라 스피너 모양을 좀 다르게 보여주는건 어떠세요? 부모에는 패딩값을 두고요!
```suggestion
          <div className="p-3 absolute inset-0 flex items-center justify-center">
```

Reply:
이 방향이 훨씬 재사용성이 높아 보이네요. 수정할게요!